src/Symfony/Component/Console/Command/InvokableCommand.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Attribute\Argument;
+use Symfony\Component\Console\Attribute\Option;
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Represents an invokable command.
+ *
+ * @author Yonel Ceruto <open@yceruto.dev>
+ *
+ * @internal
+ */
+class InvokableCommand
+{
+    private readonly \ReflectionFunction $reflection;
+
+    public function __construct(
+        private readonly Command $command,
+        private readonly \Closure $code,
+    ) {
+        $this->reflection = new \ReflectionFunction($code);
+    }
+
+    /**
+     * Invokes a callable with parameters generated from the input interface.
+     */
+    public function __invoke(InputInterface $input, OutputInterface $output): int
+    {
+        $statusCode = ($this->code)(...$this->getParameters($input, $output));
+
+        if (null !== $statusCode && !\is_int($statusCode)) {
+            // throw new LogicException(\sprintf('The command "%s" must return either void or an integer value in the "%s" method, but "%s" was returned.', $this->command->getName(), $this->reflection->getName(), get_debug_type($statusCode)));
+            trigger_deprecation('symfony/console', '7.3', \sprintf('Returning a non-integer value from the command "%s" is deprecated and will throw an exception in PHP 8.0.', $this->command->getName()));
```

Comment:
in ~PHP~ Symfony 8.0

Reply:
Fixed in https://github.com/symfony/symfony/pull/59493